### ISSUE TITLE
How to save the predicted output from LayoutLM or LayoutLMv2 ?

### ISSUE THREAD

**Issue by karndeepsingh** [microsoft/unilm#666]
I trained LayoutLM for my dataset and I am getting predictions at the word level like in the image "ALVARO FRANCISCO MONTOYA" is  true labeled as "party_name_1" but while prediction  "ALVARO " is tagged as "party_name_1", "FRANCISCO" is tagged as "party_name_1", "MONTOYA" is tagged as "party_name_1". In short, i am getting prediction for each word but how to save these prediction as one predicted output like "ALVARO FRANCISCO MONTOYA" as "party_name_1". How to save this as a single output?
Any help would be greatful.
Below image is the predicted output image from LayoutLM. 
![download (2) (2)](https://user-images.githubusercontent.com/49562460/160383498-67cccc0d-fb51-4153-bfcd-bd91bc7c7a02.png)

**Comment by ysfali** [comment]
@karndeepsingh did you find any way to tackle this problem? I'm also stuck in the same issue about how to join these results.

**Comment by karndeepsingh** [comment]
> @karndeepsingh did you find any way to tackle this problem? I'm also stuck in the same issue about how to join these results.

Nope! Still figuring it out. If you find anything do let me know.

**Comment by stjaco** [comment]
you can use IOB-like tagging: https://en.wikipedia.org/wiki/Inside%E2%80%93outside%E2%80%93beginning_(tagging)

**Comment by karndeepsingh** [comment]
> you can use IOB-like tagging: https://en.wikipedia.org/wiki/Inside%E2%80%93outside%E2%80%93beginning_(tagging)

Any reference code available to achieve it.

Thanks

**Comment by stjaco** [comment]
look at the commonly used datasets such as FUNSD / XFUND.

in your example, it boils down to training your model to recognize
`B-party_name_1` and `I-party_name_1` instead of `party_name_1`

so that the tokens 
`ALVARO FRANCISCO MONTOYA` 
will be respectively tagged as 
`B-party_name_1 I-party_name_1 I-party_name_1`

(in other words, you will know that a single `party_name_1` entity goes from `ALVARO` to `MONTOYA`)

**Comment by ysfali** [comment]
@stjaco thanks for this, I guess I'll try with BILUO based tagging. Although getting multi-line fields like addresses might still be a pain even after doing this.

**Comment by karndeepsingh** [comment]
> @stjaco thanks for this, I guess I'll try with BILUO based tagging. Although getting multi-line fields like addresses might still be a pain even after doing this.

Did you find anyway to achieve the required output?

**Comment by jyotiyadav94** [comment]
Hi @karndeepsingh ,

I am also working on something similar and I would like to ask you how did you save your predictions in a text file/csv.
Where u able to resolve this problem I am using BIOES tagging. But again getting addressed is a painful part.

**Comment by Maharshi2301** [comment]
> > @stjaco thanks for this, I guess I'll try with BILUO based tagging. Although getting multi-line fields like addresses might still be a pain even after doing this.
> 
> Did you find anyway to achieve the required output?

Any Update????
